### PR TITLE
Feat/upgrade devnet

### PR DESCRIPTION
**Pull request by hugocaillard**
### Description
Fixes: #1379 

wip

### CODE REVIEW

File: components/stacks-network/src/orchestrator.rs [hunk 1]
```diff
@@ -1263,6 +1406,180 @@ start_height = {epoch_3_0}
...
+            "7287ba251d44a4d3fd9276c88ce34c5c52a038955511cccaf77e61068649c17801",
+            "530d9f61984c888536871c6573073bdfc0058896dc1adfe9a6a10dfacadc209101",
+        ];
+
+        let signer_conf = format!(
+            r#"
+stacks_private_key = "{signer_private_key}"
+node_host = "stacks-node.{network_name}:{stacks_node_rpc_port}" # eg "127.0.0.1:20443"
+# must be added as event_observer in node config:
+endpoint = "0.0.0.0:30000" # e.g 127.0.0.1:30000
```
[the remark this answers is not in the document]
Comment (hugocaillard) [reply components/stacks-network/src/orchestrator.rs]: Not necessary but done, to have the same behavior as the regtest env

File: components/clarinet-files/src/network_manifest.rs [hunk 2]
```diff
@@ -893,6 +896,16 @@ impl NetworkManifest {
                         }
                     },
                 ),
+                stacks_signer_image_url: devnet_config
+                    .stacks_signer_image_url
+                    .take()
+                    .unwrap_or_else(|| {
+                        if devnet_config.use_nakamoto.unwrap_or(false) {
+                            DEFAULT_STACKS_NODE_IMAGE_NAKA.to_string()
```
Comment (MicaiahReid) [inline components/clarinet-files/src/network_manifest.rs:904]: Currently this won't work with the default naka image, right? Do we need to update this const to be beta2?

File: components/stacks-network/src/orchestrator.rs [hunk 3]
```diff
@@ -559,14 +597,108 @@ impl DevnetOrchestrator {
             }
         };
 
+        // Start stacks-signer-1
```
Comment (MicaiahReid) [inline components/stacks-network/src/orchestrator.rs:600]: Do we want booting the signers to be wrapped in `if use_nakamoto`?

**Review by MicaiahReid — APPROVED** [review]

**Comment by MicaiahReid** [comment]
My main concern is - do we want to disable the signer if `use_nakamoto = false`? Has this been tested with nakamoto disabled?

**Comment by hugocaillard** [comment]
@MicaiahReid
> My main concern is - do we want to disable the signer if use_nakamoto = false? Has this been tested with nakamoto disabled?

Ah yes, I might have forgot to do that 🤔 
Will do.

Testing with nakamoto disabled is part of the release playbook so I was about to 👍

File: components/stacks-network/src/orchestrator.rs (hunk 3 above)
Comment (hugocaillard) [reply components/stacks-network/src/orchestrator.rs]: Good catch

File: components/clarinet-files/src/network_manifest.rs (hunk 2 above)
Comment (hugocaillard) [reply components/clarinet-files/src/network_manifest.rs:904]: Done! Up to beta4

**Review by MicaiahReid — APPROVED** [review]

**Pull request merged**